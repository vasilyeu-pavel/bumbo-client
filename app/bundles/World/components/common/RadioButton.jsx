import React, { PropTypes } from 'react';

const RadioButton = props => (
  <div className='radio-button'>
    <input {...props} children={null} type='radio' className='radio' />
    <label htmlFor={props.id}>{props.children}</label>
  </div>
);

RadioButton.propTypes = {
  id: PropTypes.string,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  children: PropTypes.node,
};

export default RadioButton;
