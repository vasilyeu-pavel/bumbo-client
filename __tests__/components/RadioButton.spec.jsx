/* global expect, jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RadioButton from '../../app/bundles/World/components/common/RadioButton';

describe('<RadioButton /> component', () => {
  const minProps = { id: 'paymentMethod' };
  const text = 'Mastercard';

  it(`should render input type "radio" with label and text "${text}"`, () => {
    const radioButton = renderer.create(<RadioButton {...minProps} />).toJSON();

    expect(radioButton).toMatchSnapshot();
  });

  it('should call the "onChange" method after input change', () => {
    const handler = jest.fn();
    const radioButton = shallow(<RadioButton {...minProps} onChange={handler}/>);

    radioButton.find('input[type="radio"]').simulate('change');

    expect(handler).toHaveBeenCalled();
  });
});
