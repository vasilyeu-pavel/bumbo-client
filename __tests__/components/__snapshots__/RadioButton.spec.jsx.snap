exports[`<RadioButton /> component should render input type "radio" with label and text "Mastercard" 1`] = `
<div
  className="radio-button">
  <input
    className="radio"
    id="paymentMethod"
    type="radio" />
  <label
    htmlFor="paymentMethod" />
</div>
`;
